packages/ruleset/test/validate-subschemas.test.js

```diff
@@ -0,0 +1,102 @@
+const { allSchemasDocument, testRule } = require('./utils');
+const { schemas } = require('../src/collections');
+const { validateSubschemas } = require('../src/utils');
+
+describe('Utility: validateSubschemas', () => {
+  const visitedSchemas = [];
+
+  function pathRecorder(schema, path) {
+    visitedSchemas.push(path.join('.'));
+    return [];
+  }
+
+  function ruleFunction(schema, _opts, { path }) {
+    return validateSubschemas(schema, path, pathRecorder);
+  }
+
+  // this needs to be executed as a spectral rule to resolve the document
+  const ruleForTesting = {
+    given: schemas,
+    resolved: true,
+    then: {
+      function: ruleFunction
+    }
+  };
+
+  it('should find all subschemas', async () => {
+    await testRule('rule-name', ruleForTesting, allSchemasDocument);
+
+    expect(visitedSchemas[0]).toBe(
+      'paths./schema.get.responses.200.content.application/json.schema'
+    );
```

Comment:
Should we do this style instead?
```
    expect(visitedSchemas).toContain(
      'paths./schema.get.responses.200.content.application/json.schema'
    );
```

Reply:
I intentionally chose not to use that style in order to make the test say "we found these schemas _and only_ these schemas". In essence, to verify that there were no false positives in addition to no missing positives. I suppose my test still doesn't fully capture that though, because there is no check on the length of the array. I'll add that now

Reply:
Okay, sounds good!